src/config.rs

```diff
@@ -275,9 +278,16 @@ impl MiniserveConfig {
             .transpose()?
             .unwrap_or_default();
 
+        let temp_upload_directory = args.temp_upload_directory.as_ref().take().map(|v| if v.exists() && v.is_dir() {
+            Ok(v.clone())
+        } else {
+            Err(anyhow!("Upload temporary directory must exist and be a directory. Validate that path {v:?} meets those requirements"))
+        }).transpose()?;
```

Comment:
I think this would be better done at the clap level in `args.rs` with a `value_parser()` function that checks whether the path exists.